tensorflow/core/ops/array_ops.cc

```diff
@@ -2344,6 +2362,26 @@ REGISTER_OP("Slice")
         c->set_output(0, c->MakeShape(dims));
         return Status::OK();
       } else {
+        // In case `sizes` is not available, we could try to use
+        // `MakeShapeFromShapeTensor` here. If sizes contain -1, we will
+        // simply consider it as `Unknown`. This is less than ideal but
+        // still with improvement of shape inference.
+        // The following is an example that returns [None, 1, None] now
+        // (used to return [None, None, None]):
```

Comment:
change "[None, 1, None] now ... " to something like   "[None, 1, None] with this code path"

Reply:
Done.